### ISSUE TITLE
On-device debugging automatic host name setting in info.plist does not work

### ISSUE THREAD

**Issue by philipstanislaus** [facebook/react-native#11190]
### Description

When building a react-native project from Xcode (v8.1) and running it on an iPhone 6 (iOS 10.1.1), the app opens on the iPhone, but does not find a connection back to the packager on the machine. 

The developer menu (after shaking the iPhone) lacks options (the entries that are present when the phone has a connection such as "Enable Live Reload" and "Enable Hot Reloading" are not available). 
![img_3206](https://cloud.githubusercontent.com/assets/6912756/20709877/fbcd2ba8-b637-11e6-8edd-40d82c6caca2.PNG)

Tapping on "Reload" always loads from pre-bundled file, not the packager:
![img_3207](https://cloud.githubusercontent.com/assets/6912756/20709969/6c7ac7fc-b638-11e6-88f2-70fe0b889fb3.PNG)

Tapping on "Debug JS Remotely" always shows a connection refused failure: 
![img_3208](https://cloud.githubusercontent.com/assets/6912756/20710001/96163902-b638-11e6-8dda-b9db87d3c82a.PNG)

According to https://github.com/facebook/react-native/blob/master/packager/react-native-xcode.sh#L82 , the IP address of the machine with Xcode should automatically be set in plist.info. My plist.info remains unchanged, however: 

```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
  <dict>
    <key>CFBundleDevelopmentRegion</key>
    <string>en</string>
    <key>CFBundleDisplayName</key>
    <string>ByllMobile</string>
    <key>CFBundleExecutable</key>
    <string>$(EXECUTABLE_NAME)</string>
    <key>CFBundleIdentifier</key>
    <string>$(PRODUCT_BUNDLE_IDENTIFIER)</string>
    <key>CFBundleInfoDictionaryVersion</key>
    <string>6.0</string>
    <key>CFBundleName</key>
    <string>$(PRODUCT_NAME)</string>
    <key>CFBundlePackageType</key>
    <string>APPL</string>
    <key>CFBundleShortVersionString</key>
    <string>1.0</string>
    <key>CFBundleSignature</key>
    <string>????</string>
    <key>CFBundleVersion</key>
    <string>1</string>
    <key>LSRequiresIPhoneOS</key>
    <true/>
    <key>NSAppTransportSecurity</key>
    <dict>
      <key>NSExceptionDomains</key>
      <dict>
        <key>localhost</key>
        <dict>
          <key>NSExceptionAllowsInsecureHTTPLoads</key>
          <true/>
        </dict>
      </dict>
    </dict>
    <key>NSCameraUsageDescription</key>
    <string>To scan documents, please grant Byll access to your camera</string>
    <key>NSLocationWhenInUseUsageDescription</key>
    <string/>
    <key>NSPhotoLibraryUsageDescription</key>
    <string>To upload existing photos as documents, please grant Byll access to your photo library</string>
    <key>UIAppFonts</key>
    <array>
      <string>Entypo.ttf</string>
      <string>EvilIcons.ttf</string>
      <string>FontAwesome.ttf</string>
      <string>Foundation.ttf</string>
      <string>Ionicons.ttf</string>
      <string>MaterialIcons.ttf</string>
      <string>Octicons.ttf</string>
      <string>SimpleLineIcons.ttf</string>
      <string>Zocial.ttf</string>
    </array>
    <key>UILaunchStoryboardName</key>
    <string>LaunchScreen</string>
    <key>UIRequiredDeviceCapabilities</key>
    <array>
      <string>armv7</string>
    </array>
    <key>UISupportedInterfaceOrientations</key>
    <array>
      <string>UIInterfaceOrientationPortrait</string>
    </array>
    <key>UIViewControllerBasedStatusBarAppearance</key>
    <false/>
  </dict>
</plist>
```

### Solution

Automatically set the development machine's IP address in plist.info

### Additional Information

* React Native version: 0.38.0
* Platform: iOS 10.1.1
* Operating System: macOS 10.12.1

**Comment by lacker** [comment]
How does one reproduce this problem - is this happening for you just with a "hello world" install of the basic setup? Sorry but that was not clear to me from this bug description.

**Comment by philipstanislaus** [comment]
@lacker: I just created a hello world install to allow you to reproduce it: 
https://github.com/philipstanislaus/react-native-hello-world

What I do to launch the app: 

1. connect my iPhone to my MacBook
2. open `HelloWorld/ios/HelloWorld.xcodeproj` in Xcode
3. select "HelloWorld" and "Philip's iPhone" from the buttons "Set the active scheme": <img width="386" alt="bildschirmfoto 2016-12-01 um 15 51 42" src="https://cloud.githubusercontent.com/assets/6912756/20798148/21fd11ba-b7de-11e6-94e7-4f7c358c7413.png">
5. click on the play icon ("Build and then run the current scheme")

Then the packager and the app launches and the issue as described above happens (I just checked again with this HelloWorld example, same issue prevails).
